smac_planner/include/smac_planner/costmap_downsampler.hpp

```diff
@@ -162,12 +162,12 @@ class CostmapDownsampler
     for (int i = 0; i < _downsampling_factor; ++i) {
       mx = x_offset + i;
       if (mx >= _size_x) {
-        continue;
+        break;
```

Comment:
I don't think this is right, if you break then you can't get other info, I think continue was correct

Reply:
if `mx >= size_x` for the current `i` value, then increasing the `i` value further would be pointless right? 
Note that this is the SetCellCost function, that will traverse for example a 3x3 section (for downsampling_factor = 3) of the full-resolution map to assign a cost to a cell of the downsampled map. We traverse that 3x3 segment by columns, so if the current column being explored (given by `i`) is already outside of the full resolution map then there's nothing more to be explored for this specific cell and we can return.

The `downsample` method has the outer for loops that goes cell by cell of the downsampled costmap and calling that `SetCellCost` method, so we do continue assigning cost despite having the break in `SetCellCost`.

I tested this with the benchmark I've been runnning to test the downsampling and it works as expected, same as it did before the last couple of commits.

Reply:
Not necessarily. If mx > size, but my isn't, that could mean there are other rows you could look at with some partial data you're just missing.

Reply:
Maybe I'm looking at this wrongly. From the costmap code I see that mx represents the cell index along the X-axis, while my represents the cell index along the Y-axis. This means that mx is the column index and my is the row index.

If `mx > size` and you do `continue` instead of break, in the next iteration the value of mx increases by 1 and you will still have `mx > size` triggering yet another  `continue`. This would repeat until you reach the end of the for loop. 

With the assumption that the origin of the map is in the lower-left corner of the coordinate system, which seems to be true based on what I observe in rviz, there's really only three cases as far as going out-of-bounds:
1) the 3x3 window (or whatever size) is partially out-of-bounds on the right-side of the map (i.e., mx > size_x)
2) The window is partially out-of-bounds on the top-right corner of the map (i.e., mx > size_x and my > size_y)
3) The window is partially out-of-bounds on the top of the map  (i.e., my > size_y)

Your concern is valid if we traverse the window going by row first, but that's not the case. We traverse the window starting from the lower left and going upwards by column. The way we traverse the window ensures all the info is gathered in partial cells before hitting the condition mx > size. If  we hit my > size (case #3), then we break the inner loop, increase `i` and start looking at the next column (which would have relevant info)

Reply:
Please just change back to continue, this isn't worth an extended discussion over. There's edge cases this doesn't handle in at minimum far edge of map cases.